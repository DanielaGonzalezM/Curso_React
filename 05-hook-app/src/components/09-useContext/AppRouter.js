//https://stackoverflow.com/questions/63124161/attempted-import-error-switch-is-not-exported-from-react-router-dom
// In react-router-dom v6, "Switch" is replaced by routes "Routes". You need to update the import from
// import { Switch, Route } from "react-router-dom";
// to
// import { Routes ,Route } from 'react-router-dom';
// You also need to update the Route declaration from
// <Route path="/" component={Home} />
// to
// <Route path='/welcome' element={<Home/>} />

import React from 'react'
import {
    BrowserRouter as Router,
    Routes ,
    Route,
    Navigate
  } from "react-router-dom";
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NavBar from './NavBar';

const AppRouter = () => {
  return (
    <Router>
    <NavBar />

        <div>
            <Routes >
                <Route path='/about'  element={<AboutScreen/>} />
                <Route path='/login'  element={<LoginScreen/>} />
                <Route path='/'  element={<HomeScreen/>} />
                <Route path='/*' element={<Navigate replace to="/" />} />
            </Routes >
        </div>
    </Router>
    
  )
}

export default AppRouter