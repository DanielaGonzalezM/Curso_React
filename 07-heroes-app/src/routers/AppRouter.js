import React from "react";
import {
  BrowserRouter as Router,
  Routes,
  Route
} from "react-router-dom";

import LoginScreen from "../components/login/LoginScreen";

import DashboardRoutes from "./DashboardRoutes";

const AppRouter = () => {
  return (
    <div>
      <Router>
          <Routes>

            <Route path="/login" element={<LoginScreen />} />
            <Route path="/*" element={<DashboardRoutes />} />
          </Routes>
      </Router>
    </div>
  );
};

export default AppRouter;
