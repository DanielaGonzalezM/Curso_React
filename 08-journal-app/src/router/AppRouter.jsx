import { Routes, Route } from "react-router-dom";
import { AuthRoutes } from "../auth/routes/AuthRoutes";
import { JournalRoutes } from "../journal/routes/JournalRoutes";

export const AppRouter = () => {
  return (
    <Routes>
      {/* Login y registro */}
      <Route path="/auth/*" element={<AuthRoutes />} />

      {/*JournalApp */}
      <Route path="*" element={<JournalRoutes />} />
    </Routes>
  );
};
