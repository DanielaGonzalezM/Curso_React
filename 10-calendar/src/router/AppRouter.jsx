import { useEffect } from "react";
import { Navigate, Route, Routes } from "react-router-dom";
import { LoginPage } from "../auth";
import { CalendarPage } from "../calendar";
import { useAuthStore } from "../hooks";

export const AppRouter = () => {
  const { checkAuthToken, status } = useAuthStore();
  console.log(status);
  //const authStatus = "not-authenticated"; //"not-authenticated"
  useEffect(() => {
    checkAuthToken();
  }, []);

  if (status === "checking") {
    return <h3>Cargando...</h3>;
  }
  return (
    <Routes>
      {status === "not-authenticated" ? (
        <>
          <Route path="/auth/*" element={<LoginPage />} />
          <Route path="/*" element={<Navigate to="/auth/login" />} />
        </>
      ) : (
        <>
          <Route path="/" element={<CalendarPage />} />
          <Route path="/*" element={<Navigate to="/" />} />
        </>
      )}
    </Routes>
  );
};
