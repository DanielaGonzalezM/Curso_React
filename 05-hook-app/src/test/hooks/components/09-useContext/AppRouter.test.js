import { mount } from "enzyme";
import AppRouter from "../../../../components/09-useContext/AppRouter";
import UserContext from "../../../../components/09-useContext/UserContext";

describe("Pruebas AppRouter", () => {
  const user = {
    id: 1,
    name: "Daniela",
  };
  const wrapper = mount(
    <UserContext.Provider value={user}>
      <AppRouter />
    </UserContext.Provider>
  );

  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
