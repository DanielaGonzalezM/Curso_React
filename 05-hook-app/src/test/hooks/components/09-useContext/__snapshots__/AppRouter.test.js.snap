// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas AppRouter debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h3>
        HomeScreen
      </h3>
      <hr />
      <pre />
    </div>
  </div>,
]
`;
